File: OP/lab4/lab4 Python/LinearFunction.py
from TFunction import TFunction


class LinearFunction(TFunction):

    def calculate(self, x):
        return self._a * x + self._b

    def print(self):
        if self._a == 0:
            print(f"y = {self._b}", end="")
        elif self._a == 1:
            print("y = x", end="")
        elif self._a == -1:
            print("y = -x", end="")
        else:
            print(f"y = {self._a}x", end="")

        if self._b > 0:
            print(f" + {self._b}")
        elif self._b < 0:
            print(f" - {abs(self._b)}")

File: OP/lab4/lab4 C++/lab4 C++/LinearFunction.cpp
#include "LinearFunction.h"

void LinearFunction::print()
{
    if (a == 0)
        cout << "y = " << b;
    else if (a == 1)
        cout << "y = x";
    else if (a == -1)
        cout << "y = -x";
    else
        cout << "y = " << a << "x";

    if (b > 0)
        cout << " + " << b;
    else if (b < 0)
        cout << " - " << abs(b);
    cout << endl;
}

double LinearFunction::calculate(double x)
{
    return a * x + b;
}